dashboard/app/models/plc/course_unit.rb

```diff
@@ -34,22 +34,17 @@ def get_required_learning_module_ids
     plc_learning_modules.required.pluck(:id)
   end
 
-  def get_top_modules_of_each_type_from_user_selections(learning_module_ids)
+  def get_top_modules_of_each_type_from_user_selections(learning_module_ids_and_weights)
     # Preload all of the learning modules, that way we don't have to hit the DB multiple times
-    learning_module_ids.compact!
-
-    return [] if learning_module_ids.nil? || learning_module_ids.empty?
-    learning_module_map = Hash[*Plc::LearningModule.find(learning_module_ids).flat_map {|learning_module| [learning_module.id, learning_module]}]
-
-    #This will make a hash map of learning_module => number of times the user picked it in the evaluation, sorted with the
-    #most frequently selected ones first
-    searching_map = Hash[*learning_module_ids.group_by{ |v| v }.map{ |k, v| [learning_module_map[k.to_i], v.size] }.sort_by { |x| x[1] }.reverse.flatten]
+    return [] if learning_module_ids_and_weights.nil? || learning_module_ids_and_weights.empty?
```

Comment:
This can be done with .blank? (since it's in rails), and in general ruby with .to_a == []

```ruby
irb(main):001:0> [].blank?
=> true
irb(main):002:0> nil.blank?
=> true
irb(main):003:0> [].to_a == []
=> true
irb(main):004:0> nil.to_a == []
=> true
```